### PR TITLE
[UI/UX] Use Tabs on Game Settings

### PR DESCRIPTION
**Pull request by flavioislima**
This replaces the Collapsable elements with a Tabbing approach on the Game Settings.
It also remembers the last used tab for each game so makes our life easier when testing different configs.

image

image

image

image

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/frontend/screens/Settings/components/SettingsModal/index.scss [hunk 1]
```diff
@@ -2,8 +2,8 @@
   width: 65vw;
   display: flex;
   flex-direction: column;
-  justify-content: space-between;
   max-width: 800px;
+  min-height: 50vh;
```
Comment (arielj) [inline src/frontend/screens/Settings/components/SettingsModal/index.scss:6]: maybe we can also position the modal a fixed amount from the top? now changing tabs changes the height and it keeps jumping around, it's not great for accessibility that much movement

File: src/frontend/screens/Settings/sections/GamesSettings/index.tsx [hunk 2]
```diff
@@ -82,101 +118,75 @@ export default function GamesSettings({ useDetails = true }: Props) {
...
-
-        <WrappersTable />
+      <TabPanel value={value} index={1}>
+        {!isCrossover && (
+          <>
+            <GameMode />
+            <ShowFPS />
+            <Mangohud />
+            <BattlEyeRuntime />
+            <EacRuntime />
```
Comment (arielj) [inline src/frontend/screens/Settings/sections/GamesSettings/index.tsx:164]: this is showing on Mac but I understand EAC doesn't work on Mac (I haven't checked yet if it's an issue also on main though)

Comment (Etaash-mathamsetty) [reply src/frontend/screens/Settings/sections/GamesSettings/index.tsx:164]: both runtimes are linux exclusive

**Comment by arielj** [comment]
Looks like the change to a different place broke the alignment for the text of these buttons

Screenshot 2023-10-01 at 4 22 59 PM

**Comment by arielj** [comment]
Going to the General Settings > Game Defaults crashes with:
Screenshot 2023-10-01 at 4 25 14 PM

I guess it's for the tab remembering thingy?

File: src/frontend/screens/Settings/sections/GamesSettings/index.tsx (hunk 2 above)
Comment (flavioislima) [reply src/frontend/screens/Settings/sections/GamesSettings/index.tsx]: fixed

**Comment by flavioislima** [comment]
> Looks like the change to a different place broke the alignment for the text of these buttons
> Screenshot 2023-10-01 at 4 22 59 PM

Fixed and Also added a small fading animation to when they are running:
image

**Comment by flavioislima** [comment]
> Going to the General Settings > Game Defaults crashes with: Screenshot 2023-10-01 at 4 25 14 PM
> 
> I guess it's for the tab remembering thingy?

Fixed that as well.

**Comment by imLinguin** [comment]
obraz

Let's make Wine optional tab for native games :D

**Comment by flavioislima** [comment]
> obraz
> 
> Let's make Wine tab hidden for native games :D
> 
> Same goes for some options in different tabs, like anti cheat runtimes
> 
> obraz

Fixed.
Also changed the index to strings because using numbers was causing other issues when the Wine tab was not available.

File: src/frontend/screens/Settings/components/SettingsModal/index.scss (hunk 1 above)
Comment (flavioislima) [reply src/frontend/screens/Settings/components/SettingsModal/index.scss:6]: good idea, that was bothering me as well. I will try.

File: src/frontend/screens/Settings/components/Tools/index.tsx [hunk 3]
```diff
@@ -98,6 +105,12 @@ export default function Tools() {
     ev.preventDefault()
   }
 
+  console.log('toolsSettings', {
+    winetricksRunning,
+    winecfgRunning,
+    runExeRunning
+  })
```
Comment (arielj) [inline src/frontend/screens/Settings/components/Tools/index.tsx:112]:
```suggestion
```

Comment (arielj) [reply src/frontend/screens/Settings/components/Tools/index.tsx:112]: to delete this

File: src/frontend/screens/Settings/components/Tools/index.tsx [hunk 4]
```diff
@@ -131,12 +144,15 @@ export default function Tools() {
...
+            })}
             onClick={handleRunExe}
           >
-            {t('setting.runexe.title')}
-            <br />
-            <span>{t('setting.runexe.message')}</span>
+            <div>
+              {t('setting.runexe.title')} <br />
+              <span>{t('setting.runexe.message')}</span>
+            </div>
```
Comment (arielj) [inline src/frontend/screens/Settings/components/Tools/index.tsx:155]: we don't need this div, we can also remove the new css, we need to add a `justify-content: center` to `.drag` fix the alignment of the run exe on prefix button

File: src/frontend/screens/Settings/components/Tools/index.scss [hunk 5]
```diff
@@ -20,7 +27,12 @@
     text-decoration: none;
     white-space: nowrap;
 
-    & > span {
+    & div {
+      position: relative;
+      top: -18px;
+    }
```
Comment (arielj) [inline src/frontend/screens/Settings/components/Tools/index.scss:33]:
```suggestion
    justify-content: center;
```

(I left another comment to remove the div, and with this css it fixes the alingment)

File: src/frontend/screens/Settings/sections/GamesSettings/index.tsx [hunk 6]
```diff
@@ -82,101 +116,81 @@ export default function GamesSettings({ useDetails = true }: Props) {
...
+        )}
+        {hasCloudSaves && (
+          <Tab
+            label={t('settings.navbar.sync', 'Cloud Saves Sync')}
+            value="saves"
+          />
+        )}
+      </Tabs>
+
+      <TabPanel value={value} index={'wine'}>
```
Comment (arielj) [inline src/frontend/screens/Settings/sections/GamesSettings/index.tsx:136]: we can remove the fragment after this line now

**Review by arielj — COMMENTED** [review]
I left a few comments, but it works good for me and looks way better.

I'll wait for the change setting a fixed top distance and it's all good from me

File: src/frontend/screens/Settings/components/SettingsModal/index.scss (hunk 1 above)
Comment (flavioislima) [reply src/frontend/screens/Settings/components/SettingsModal/index.scss:7]: done!

File: src/frontend/screens/Settings/sections/GamesSettings/index.tsx (hunk 6 above)
Comment (flavioislima) [reply src/frontend/screens/Settings/sections/GamesSettings/index.tsx:141]: fixed

File: src/frontend/screens/Settings/components/Tools/index.scss (hunk 5 above)
Comment (flavioislima) [reply src/frontend/screens/Settings/components/Tools/index.scss]: fixed

File: src/frontend/screens/Settings/components/Tools/index.tsx (hunk 4 above)
Comment (flavioislima) [reply src/frontend/screens/Settings/components/Tools/index.tsx]: fixed as well

**Comment by flavioislima** [comment]
Addressed all comments, I think looks good now.

**Review by arielj — APPROVED** [review]
Looks good to me :+1:

**Review by arielj — CHANGES_REQUESTED** [review]
wait, I found a bug opening settings for a browser app sideloaded 

image

**Comment by arielj** [comment]
(well I guess we have a bigger problem with those apps, we are showing a bunch of settings that makes no sense, maybe the settings button for sideloaded apps should open the edit app modal instead of settings, no setting applies for theme)

**Comment by arielj** [comment]
That same error happens for any native linux game since the `wine` tab doesn't exist. I imagine it's the same for native Mac games too

**Comment by arielj** [comment]
when that happens, there's no `Wine` tab, but the wine tab content is displayed 
image

**Comment by flavioislima** [comment]
@arielj ooff, the burden of supporting so many configurations I guess. lol
will check, thanks for testing!

**Comment by flavioislima** [comment]
@arielj removed the settings button for browser games for now until we have proper settings for them (maybe user-agent, open fullscreen by default, etc) in the future.
Improved the logic to get more use cases and tested on macOS, Windows and Linux, native, sideloaded, wine, etc.
Also the Default game settings on each platform
Could not find any other issue so far.

**Review by arielj — APPROVED** [review]
Looks good to me :+1: 

I tested only on linux though, but I haven't seen any other issue.

**Pull request merged**